diff --git a/coreutils/df.c b/coreutils/df.c
index 63dbd61..4af2d9e 100644
--- a/coreutils/df.c
+++ b/coreutils/df.c
@@ -68,6 +68,21 @@ static unsigned long kscale(unsigned long b, unsigned long bs)
 }
 #endif
 
+struct mntent *_getmntent (FILE *);
+struct mntent *_getmntent (FILE *stream)
+{
+  static struct mntent m;
+  static char *getmntent_buffer;
+
+  #define BUFFER_SIZE 4096
+  if (getmntent_buffer == NULL) {
+    getmntent_buffer = (char *) malloc (BUFFER_SIZE);
+  }
+
+  return getmntent_r (stream, &m, getmntent_buffer, BUFFER_SIZE);
+  #undef BUFFER_SIZE
+}
+
 int df_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int df_main(int argc UNUSED_PARAM, char **argv)
 {
@@ -150,7 +165,7 @@ int df_main(int argc UNUSED_PARAM, char **argv)
 		const char *mount_point;
 
 		if (mount_table) {
-			mount_entry = getmntent(mount_table);
+			mount_entry = _getmntent(mount_table);
 			if (!mount_entry) {
 				endmntent(mount_table);
 				break;
